client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx

```diff
@@ -11,24 +12,21 @@ import Button from 'components/button';
 import DisconnectJetpackDialog from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog';
 import analytics from 'lib/analytics';
 
-export default React.createClass( {
+class DisconnectJetpackButton extends Component {
+	constructor( props ) {
+		super( props );
 
-	displayName: 'DisconnectJetpackButton',
-
-	propTypes: {
-		site: React.PropTypes.object.isRequired,
-		redirect: React.PropTypes.string.isRequired,
-		disabled: React.PropTypes.bool,
-		linkDisplay: React.PropTypes.bool,
-		isMock: React.PropTypes.bool,
-		text: React.PropTypes.string
-	},
+		this.handleClick = this.handleClick.bind( this );
+	}
 
-	getDefaultProps() {
-		return {
-			linkDisplay: true
-		};
-	},
+	handleClick( event ) {
+		event.preventDefault();
+		if ( this.props.isMock ) {
+			return;
+		}
+		this.refs.dialog.getWrappedInstance().open();
```

Comment:
Good point - adding a check now, and we can revisit to remove that ref in a future PR.